### PR TITLE
Add ThreeTickTeaks Plugin, Add name to PluginDescriptor

### PR DESCRIPTION
**Pull request by ghost**
Woohooo first contribution to a open source project! :)

I have tested it out a bit and it seems to work good so far, got a woodcutting level with it!

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/eeng1n/woodcutting/threetickteaks/ThreeTickTeaks.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+    @Provides
+    ThreeTickTeaksConfig getConfig(ConfigManager manager) {
+        return manager.getConfig(ThreeTickTeaksConfig.class);
+    }
+
+    @Override
+    protected void startUp() {
+        if (Microbot.getClient().getGameState() == GameState.LOGGED_IN) {
+            Microbot.pauseAllScripts = false;
```
Comment (chsami) [inline runelite-client/src/main/java/net/runelite/client/plugins/eeng1n/woodcutting/threetickteaks/ThreeTickTeaks.java]: Line 70 to 74 can be removed as they are redundant.

These setters are set in the MicrobotPlugin.java which is the main plugin enabled by default

File: runelite-client/src/main/java/net/runelite/client/plugins/eeng1n/woodcutting/threetickteaks/ThreeTickTeaks.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+@PluginDescriptor(
+        name = PluginDescriptor.Engin + " 3T Teaks",
+        description = "Performs 3T Teaks",
+        enabledByDefault = false
+)
+@Slf4j
+public class ThreeTickTeaks extends Plugin {
+    @Inject
+    private Client client;
```
Comment (chsami) [inline runelite-client/src/main/java/net/runelite/client/plugins/eeng1n/woodcutting/threetickteaks/ThreeTickTeaks.java]: The following injects can be removed
-client
-clientThread
-notifier

these are injected and set in the MicrobotPlugin.java

File: runelite-client/src/main/java/net/runelite/client/plugins/eeng1n/woodcutting/threetickteaks/ThreeTickTeaks.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    ThreeTickTeaksState state = ThreeTickTeaksState.Idle;
+
+    @Provides
+    ThreeTickTeaksConfig getConfig(ConfigManager manager) {
+        return manager.getConfig(ThreeTickTeaksConfig.class);
+    }
+
+    @Override
+    protected void startUp() {
+        if (Microbot.getClient().getGameState() == GameState.LOGGED_IN) {
```
Comment (chsami) [inline runelite-client/src/main/java/net/runelite/client/plugins/eeng1n/woodcutting/threetickteaks/ThreeTickTeaks.java:59]: You should add an extra check to see if the player has the required items before starting. 

-guam
-pestle and mortar
etc...

Also add an extra check to see if the player has the required woodcutting level before starting the script.

 you can use something like  Microbot.showMessage to display the message.

File: runelite-client/src/main/java/net/runelite/client/plugins/eeng1n/woodcutting/threetickteaks/ThreeTickTeaksConfig.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+@ConfigGroup("threeTickTeaks")
+public interface ThreeTickTeaksConfig extends Config {
+    @ConfigItem(
+            keyName = "guide",
+            name = "How to use",
+            description = "How to use this plugin",
+            position = 0
+    )
+    default String GUIDE() {
+        return "1. Have your inventory setup for 3T Teaks \n2. Enable Plugin";
```
Comment (chsami) [inline runelite-client/src/main/java/net/runelite/client/plugins/eeng1n/woodcutting/threetickteaks/ThreeTickTeaksConfig.java]: It's always nice to sum up the item requirements and stat requirements and location requirement for the script

File: runelite-client/src/main/java/net/runelite/client/plugins/eeng1n/woodcutting/threetickteaks/ThreeTickTeaks.java [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+import java.util.concurrent.Executors;
+
+import static net.runelite.client.plugins.microbot.util.Global.sleep;
+import static net.runelite.client.plugins.microbot.util.Global.sleepUntilOnClientThread;
+import static net.runelite.client.plugins.natepainthelper.Info.*;
+
+@PluginDescriptor(
+        name = PluginDescriptor.Engin + " 3T Teaks",
+        description = "Performs 3T Teaks",
+        enabledByDefault = false
```
Comment (chsami) [inline runelite-client/src/main/java/net/runelite/client/plugins/eeng1n/woodcutting/threetickteaks/ThreeTickTeaks.java:32]: Please add tags for your script, this makes it easier to find when people are searching for "woodcutting"

example of tags:

tags = {"tithe farming", "microbot", "skills", "minigame"},

**Review by chsami — CHANGES_REQUESTED** [review]
small QOL changes for the script

**Comment by ghost** [comment]
Should be enough QOL implemented (Only the How to use Section is still bugged for me)

**Review by chsami — APPROVED** [review]

**Pull request merged**